locallib.php

```diff
@@ -477,19 +477,40 @@ public function get_form_elements_for_user($submission, MoodleQuickForm $mform,
                 );
 
                 foreach ($views['data'] as $view) {
-                    $viewurl = "/view/view.php?id=" . $view['id'];
-                    $anchor = $this->get_preview_url($view['title'], $viewurl, strip_tags($view['description']));
-                    $mform->addElement('radio', 'viewid', '', $anchor, 'v' . $view['id']);
+                    if ($view['submissionoriginal'] == 0) {
+                        $viewurl = "/view/view.php?id=" . $view['id'];
+                        $anchor = $this->get_preview_url($view['title'], $viewurl, strip_tags($view['description']));
+                        $mform->addElement('radio', 'viewid', '', $anchor, 'v' . $view['id']);
+                    }
+                    if ($maharasubmission && $view['id'] == $maharasubmission->viewid) {
+                        $currentpagesubmitted = $mform->createElement('static', 'currentsubmission',
+                            get_string('currentsubmitted', 'assignsubmission_maharaws', 'page'), $view['displaytitle']);
```

Comment:
Looks like this can can be abstracted to a method to avoid a code duplication. See pretty much same code at lines 506-507

Reply:
I've run out of time on this WR so can only do what's necessary to move it along I'm afraid.